### PR TITLE
My solution task web security

### PR DESCRIPTION
**Pull request by Swyatik19**

### CODE REVIEW

File: src/main/java/taxi/controller/car/GetAllCarsController.java [hunk 1]
```diff
@@ -18,6 +18,8 @@ public class GetAllCarsController extends HttpServlet {
     public void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         List<Car> cars = carService.getAll();
+        String h1 = "My all cars";
+        req.setAttribute("h1", h1);
```
Comment (Rommelua) [inline src/main/java/taxi/controller/car/GetAllCarsController.java:22]: Why do you need this? You have code duplication here and in GetMyCurrentCarsController
It should be on jsp

File: src/main/java/taxi/service/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password was incorrect.");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
```
Comment (Rommelua) [inline src/main/java/taxi/service/AuthenticationServiceImpl.java:22]: Both in AuthenticationServiceImpl and in AuthenticationFilter, use only one if block if possible.
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-web/web-security/java-web-security)

File: src/main/java/taxi/web/filter/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        Long driverId = (Long) session.getAttribute("driver_id");
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null) {
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
```
Comment (Rommelua) [inline src/main/java/taxi/web/filter/AuthenticationFilter.java:40]: Both in AuthenticationServiceImpl and in AuthenticationFilter, use only one if block if possible.
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-web/web-security/java-web-security)

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/controller/car/GetMyCurrentCarsController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("taxi");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long driverId = (Long) req.getSession().getAttribute("driver_id");
+        List<Car> carsById = carService.getAllByDriver(driverId);
+        String h1 = "My all cars";
+        req.setAttribute("h1", h1);
```
Comment (Rommelua) [inline src/main/java/taxi/controller/car/GetMyCurrentCarsController.java:23]: this text should be on jsp page - do not pass it as attribute

Comment (Swyatik19) [reply src/main/java/taxi/controller/car/GetMyCurrentCarsController.java]: Sorry, I forgot to check the second class

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**